Source/items.cpp

```diff
@@ -2169,10 +2169,7 @@ void SpawnOnePremium(Item &premiumItem, int plvl, int playerId)
 
 	plvl = clamp(plvl, 1, 30);
 
-	int count = 0;
-
-	do {
-		keepGoing = false;
+	for (int count = 0; count < 150; count++) {

```

Comment:
Question: Is this the same as master? And does it matter?
In master and diablo (`gbIsHellfire` = false) the loop could iterates more then 150 times if `premiumItem._iIvalue > 140000` is always true.

Reply:
Good catch, just checked Devilution. I guess hypothetically you could be stuck in the loop forever if you got very unlucky? lol

Could separate the increment out of the loop declaration and just increment as part of the hellfire side - or set count to 0 when entering the Diablo side (the former seems less "hacky" to me). 

I guess the question is how long you could get realistically stuck in the loop, and does it make sense to have a count limitation in general, even if it didn't work like that originally? Any thoughts here?

Reply:
> Good catch, just checked Devilution. I guess hypothetically you could be stuck in the loop forever if you got very unlucky? lol
> 
> Could separate the increment out of the loop declaration and just increment as part of the hellfire side - or set count to 0 when entering the Diablo side (the former seems less "hacky" to me).
> 
> I guess the question is how long you could get realistically stuck in the loop, and does it make sense to have a count limitation in general, even if it didn't work like that originally? Any thoughts here?

This is true, in fact you can get stuck in the loading screen if this function never resolves. I don't think this is ideal.